transaction/src/validation/id_validator.rs

```diff
@@ -119,7 +119,10 @@ impl ManifestValidator {
     }
 
     pub fn drop_all_named_proofs(&mut self) -> Result<(), ManifestIdValidationError> {
-        self.proof_ids.clear();
```

Comment:
We had a bug here where calling `self.proof_ids.clear();` never reduced the bucket's ref count. Thus, creating a proof from a bucket and then calling `drop_all_named_proofs` would produce manifests that fail static validation and be rejected.

Reply:
Nice spot!